airflow/models/dag.py

```diff
@@ -3505,3 +3556,57 @@ def get_current_dag(cls) -> DAG | None:
             return cls._context_managed_dags[0]
         except IndexError:
             return None
+
+
+def _run_task(ti: TaskInstance, session):
+    """
+    Run a single task instance, and push result to Xcom for downstream tasks. Bypasses a lot of
+    extra steps used in `task.run` to keep our local running as fast as possible
+    Args:
+        ti: TaskInstance to run
+    """
+    log.info("*****************************************************")
+    if ti.map_index > 0:
+        log.info("Running task %s index %d", ti.task_id, ti.map_index)
+    else:
+        log.info("Running task %s", ti.task_id)
+    try:
+        ti._run_raw_task(session=session)
+        session.flush()
+        log.info("%s ran successfully!", ti.task_id)
+    except AirflowSkipException:
+        log.info("Task Skipped, continuing")
+    log.info("*****************************************************")
+    ti.set_state(State.SUCCESS)
```

Comment:
so maybe what we can do here ... is instead of always setting success here... just look at what the state of the TI is after it runs.... (the method already sets state, i believe!)

and... in the special case of a task getting deferred.... maybe ti would be best to  just fail it right then and there.... and .... you could do so at least _after the fact_ by inspecting ti.state.

Reply:
@dstandish yep I now just let _run_raw_task do the work here.